src-self-hosted/Module.zig

```diff
@@ -2683,7 +2683,56 @@ fn analyzeInstElemPtr(self: *Module, scope: *Scope, inst: *zir.Inst.ElemPtr) Inn
 }
 
 fn analyzeInstSub(self: *Module, scope: *Scope, inst: *zir.Inst.Sub) InnerError!*Inst {
-    return self.fail(scope, inst.base.src, "TODO implement analysis of sub", .{});
+    const lhs = try self.resolveInst(scope, inst.positionals.lhs);
```

Comment:
This implementation looks like it was copied from add and then modified right? I think this is a part where we would actually be well-advised to port fairly directly from stage1, which has this:

```c++
        case IrBinOpBinOr:
        case IrBinOpBinXor:
        case IrBinOpBinAnd:
        case IrBinOpAdd:
        case IrBinOpAddWrap:
        case IrBinOpSub:
        case IrBinOpSubWrap:
        case IrBinOpMult:
        case IrBinOpMultWrap:
        case IrBinOpDivUnspecified:
        case IrBinOpDivTrunc:
        case IrBinOpDivFloor:
        case IrBinOpDivExact:
        case IrBinOpRemUnspecified:
        case IrBinOpRemRem:
        case IrBinOpRemMod:
            return ir_analyze_bin_op_math(ira, bin_op_instruction);
```

If you look at the implementation of `ir_analyze_bin_op_math`, there is a lot of logic there that can be applied to all these operations.

This PR does not need to handle all these operations, but it should at least have the same rough skeleton as that code, in that the logic between add and subtract is shared.

Reply:
Thanks for the feedback, I'll work on this tomorrow.